core/src/main/java/org/geysermc/geyser/session/GeyserSession.java

```diff
@@ -1427,7 +1427,7 @@ public void requestOffhandSwap() {
 
     @Override
     public String name() {
-        return null;
+        return this.isLoggedIn() ? this.javaUsername() : this.bedrockUsername();
```

Comment:
`Session#disconnect(reason)` will set `loggedIn` to false, and then fire the `SessionDisconnectEvent`.

Maybe we call `javaUsername()`, and if it is null, fall back to `bedrockUsername()`?